packages/astro/src/jsx/server.ts

```diff
@@ -38,6 +51,22 @@ export async function renderToStaticMarkup(
 	return { html };
 }
 
+type FormatErrorOptions = {
+	message: string;
+	name: string;
+	stack?: string;
+	hint: string;
+	title: string;
+};
+function createFormattedError({ message, name, stack, hint }: FormatErrorOptions) {
+	const error = new Error(message);
+	error.name = name;
+	error.stack = stack;
+	// @ts-expect-error - hint is not part of the Error interface but it will be picked up by the error overlay
+	error.hint = hint;
+	return error;
+}
+
```

Comment:
You're looking for the AstroError class

Reply:
Ah, I had assumed that it was meant to be used exclusively for errors related to Astro's core features. I guess this should be categorized as a `MarkdownError`?

Reply:
I reviewed this again but I don't think `AstroError` should be used. This function is just used to format a standard JavaScript runtime error message, which could be any type of error (`ReferenceError`, `SyntaxError`, ...).

For example, we don't use `AstroError` for runtime errors in Astro components or in framework components, we just pull them from the vite overlay.

Naming this would be confusing too. Should we display the name of the `MdxRuntimeError`, the name of the actual error (`ReferenceError` for example), or both?